sycl/plugins/opencl/CMakeLists.txt

```diff
@@ -26,14 +33,37 @@ set_target_properties(pi_opencl PROPERTIES LINKER_LANGUAGE CXX)
 #preprocessor definitions for compiling a target's sources. We do not need it for pi_opencl
 target_include_directories(pi_opencl PRIVATE "${sycl_inc_dir}")
 
-#link pi_opencl with OpenCL headers and ICD Loader.
-target_link_libraries( pi_opencl
+if (MSVC)
+  target_compile_definitions(pi_opencl PRIVATE __SYCL_BUILD_SYCL_DLL)
+  target_link_libraries(pi_opencl PRIVATE shlwapi)
+  target_link_libraries( pi_opencl
     PRIVATE OpenCL::Headers
     PRIVATE ${OpenCL_LIBRARIES}
-)
+  )
+else()
+  # we set the visibility of all symbols 'hidden' by default.
+  # In *.cpp files, we set exported symbols with visibility==default individually
+  target_compile_options(pi_opencl PUBLIC -fvisibility=hidden)
+
+  # This script file is used to allow exporting pi* symbols only.
+  # All other symbols are regarded as local (hidden)
+  set(linker_script "${CMAKE_CURRENT_SOURCE_DIR}/../ld-version-script.txt")
+
+  #link pi_opencl with OpenCL headers and ICD Loader.
+  target_link_libraries( pi_opencl
```

Comment:
Please, avoid code duplication.

Reply:
This script is not the same as libsycl.so uses.
We export only pi* symbols, and hide all others.

Reply:
I’m talking about two branches of this if-else statement. You can factor out duplicated libs (those are opencl and headers) and link them unconditionally.

Reply:
Thanks.